### PR TITLE
TaskHandle now returns (bool, error)

### PR DESCRIPTION
**Pull request by skonefal**
Merging Snap SessionTaskHandle &  Exector TaskHandle.

Executor's TaskHandle now returns (bool, error).

### CODE REVIEW

**Comment by skonefal** [comment]
run integration tests

**Comment by skonefal** [comment]
run integration tests

File: integration_tests/pkg/executor/kubernetes_test.go [hunk 1]
```diff
@@ -149,7 +153,9 @@ func TestKubernetesExecutor(t *testing.T) {
 			defer executor.StopAndEraseOutput(taskHandle)
 
 			So(err, ShouldBeNil)
-			So(taskHandle.Wait(podFinishedTimeout), ShouldBeTrue)
+			terminated, err := taskHandle.Wait(0)
```
Comment (iwankgb) [inline integration_tests/pkg/executor/kubernetes_test.go]: @skonefal - see question in lint 127, please.

File: pkg/executor/cluster_task_handle.go [hunk 2]
```diff
@@ -86,20 +86,33 @@ func (m *ClusterTaskHandle) ExitCode() (int, error) {
 // Wait does the blocking wait for the master completion in case of 0 timeout time.
 // Wait is a helper for waiting with a given timeout time.
 // It returns true if all tasks are terminated.
-func (m *ClusterTaskHandle) Wait(timeout time.Duration) bool {
+func (m *ClusterTaskHandle) Wait(timeout time.Duration) (isMasterTerminated bool, err error) {
 	// Wait for master first.
-	isMasterTerminated := m.master.Wait(timeout)
+	isMasterTerminated, err = m.master.Wait(timeout)
+	if err != nil {
+		logrus.Errorf("Waiting for %q failed: %s", m.master.Name(), err)
```
Comment (ppalucki) [inline pkg/executor/cluster_task_handle.go]: drop logging if we return errors (leave error handling for upper layers)

File: pkg/executor/cluster_task_handle.go [hunk 3]
```diff
@@ -86,20 +86,33 @@ func (m *ClusterTaskHandle) ExitCode() (int, error) {
...
+	return isMasterTerminated, err
+}
+
+func (m *ClusterTaskHandle) stopAgents() error {
+	var errCol errcollection.ErrorCollection
+	// Stop the agents when master is terminated.
+	for _, handle := range m.agents {
+		err := handle.Stop()
+		if err != nil {
+			logrus.Errorf("Error while stopping %q: %s", handle.Name(), err)
```
Comment (ppalucki) [inline pkg/executor/cluster_task_handle.go]: drop logging if we return errors (leave error handling for upper layers)

**Comment by skonefal** [comment]
Thanks @iwankgb @ppalucki !

**Comment by skonefal** [comment]
run integration tests

**Comment by skonefal** [comment]
run integration tests

**Pull request merged**